Tests/fixtures/clienthints.yml

```diff
@@ -0,0 +1,662 @@
+    engine: Blink
+    engine_version: ""
+  device:
+    type: desktop
+    brand: ""
+    model: Xbox
+  os_family: Windows
+  browser_family: Internet Explorer
+-
+  user_agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.16 Safari/537.36 OPR/85.0.4338.0 (Edition developer)
+  headers:
+    Sec-CH-UA: '" Not A;Brand";v="99.0.0.0", "Chromium";v="99.0.4844.16", "Opera Crypto";v="99.0.4844.16"'
+    Sec-CH-UA-Platform: "Windows"
+    Sec-CH-UA-Mobile: "?0"
+    Sec-CH-UA-Full-Version: "99.0.4844.16"
+    Sec-CH-UA-Platform-Version: "10.0.0"
+    Sec-CH-UA-Arch: "x86"
+    Sec-CH-UA-Bitness: "64"
+    Sec-CH-Prefers-Color-Scheme: "dark"
+  os:
+    name: Windows
+    version: "10"
+    platform: x64
+  client:
+    type: browser
+    name: Opera Crypto
+    version: "99.0.4844.16"
+    engine: Blink
+    engine_version: ""
+  device:
+    type: desktop
+    brand: ""
+    model: ""
+  os_family: Windows
+  browser_family: Opera
+-
+  user_agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.109 Safari/537.36
+  headers:
+    Sec-CH-UA: '" Not A;Brand";v="99.0.0.0", "Chromium";v="98.0.4758.109", "Gener8";v="98.0.4758.109"'
+    Sec-CH-UA-Platform: "Windows"
+    Sec-CH-UA-Mobile: "?0"
+    Sec-CH-UA-Full-Version: "98.0.4758.109"
+    Sec-CH-UA-Platform-Version: "10.0.0"
+    Sec-CH-UA-Arch: "x86"
+    Sec-CH-UA-Bitness: "64"
+    Sec-CH-Prefers-Color-Scheme: "light"
+  os:
+    name: Windows
+    version: "10"
+    platform: x64
+  client:
+    type: browser
+    name: Gener8
+    version: "98.0.4758.109"
+    engine: Blink
+    engine_version: ""
+  device:
+    type: desktop
+    brand: ""
+    model: ""
+  os_family: Windows
+  browser_family: Chrome
+-
+  user_agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36 PTST/220216.200657
+  headers:
+    Sec-CH-UA-Platform: "Unknown"
+    Sec-CH-UA-Mobile: "?0"
+  bot:
+    name: WebPageTest
+    category: Site Monitor
+    url: https://www.webpagetest.org
+-
+  user_agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36 PTST/220216.200657
+  headers:
+    Sec-CH-UA: '" Not A;Brand";v="99", "Chromium";v="98.0.4758.102", "Google Chrome";v="98.0.4758.102"'
+    Sec-CH-UA-Platform: "Unknown"
+    Sec-CH-UA-Mobile: "?0"
+    Sec-CH-UA-Full-Version: "98.0.4758.102"
+    Sec-CH-UA-Platform-Version: "0"
+    Sec-CH-UA-Arch: "x86"
+    Sec-CH-UA-Model: "Model"
+    Sec-CH-Prefers-Color-Scheme: "light"
+  bot:
+    name: WebPageTest
+    category: Site Monitor
+    url: https://www.webpagetest.org
+-
+  user_agent: Mozilla/5.0 (X11; CrOS x86_64 14150.90.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.126 Safari/537.36
+  headers:
+    Sec-CH-UA-Platform: "Chromium OS"
+    Sec-CH-UA-Mobile: "?0"
+  os:
+    name: Chromium OS
+    version:
+    platform: x64
```

Comment:
Here, version is not detected.

Reply:
And, regarding Opera, it should get the version from user agent, rather than client hints, since that is the real version.

Reply:
Ok. changed the code, so the version from useragent is used if none is provided, but the os family matches.
We need to see if that might cause any incorrect detections in the future...